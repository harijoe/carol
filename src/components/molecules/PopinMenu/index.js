import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { List, ListItem } from 'components'
import theme from '../../themes/default'

const Background = styled.div`
  display: ${props => props.display ? 'block' : 'none'};
  position: fixed;
  z-index: 3;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
`

const StyledButton = styled.button`
  background: inherit;
  border: none;
  z-index: 4;
  position: relative;

  &:focus {
    outline: none;
  }
`

const Menu = styled.div`
  transition: visibility 0.2s, opacity 0.2s linear;
  opacity: ${props => props.display ? '1' : '0'};
  position: absolute;
  z-index: 5;
  background-color: ${theme.colors.white};
  bottom: 50px;
  box-shadow: 0 0 40px 0 rgba(19, 19, 19, 0.2);
`

const StyledList = styled(List)`
  padding: 0 20px;
  list-style: none;
`

const StyledItem = styled(ListItem)`
  padding: 0;
  border-bottom: 1px solid ${theme.colors.grayscale.medium};

  &:last-child {
    border-bottom: none;
  }
`

class PopinMenu extends Component {
  static propTypes = {
    children: PropTypes.any.isRequired,
    menu: PropTypes.array.isRequired,
  }

  state = {
    display: false,
  }

  toggle = () => {
    this.setState({ display: !this.state.display })
  }

  hide = () => {
    this.setState({ display: false })
  }

  render() {
    const { menu, children } = this.props

    return (
      <div>
        <Background display={this.state.display} onClick={this.hide} />
        <Menu display={this.state.display}>
          <StyledList>
            {
              menu.map((element, i) => (
                <StyledItem
                  key={i}
                  onClick={this.hide}
                >
                  {element}
                </StyledItem>
              ))
            }
          </StyledList>
        </Menu>
        <StyledButton type="button" onClick={this.toggle}>
          {children}
        </StyledButton>
      </div>
    )
  }
}

export default PopinMenu
