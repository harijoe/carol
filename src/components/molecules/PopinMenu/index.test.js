import React from 'react'
import { shallow } from 'enzyme'

import PopinMenu from './'

it('renders PopinMenu', () => {
  const wrapper = shallow(<PopinMenu menu={['hello']}>Test</PopinMenu>)

  expect(wrapper).toMatchSnapshot()
})
