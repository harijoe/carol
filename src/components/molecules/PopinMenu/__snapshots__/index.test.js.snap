// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PopinMenu 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopinMenu
    menu={
      Array [
        "hello",
      ]
    }
  >
    Test
  </PopinMenu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Background
          display={false}
          onClick={[Function]}
        />,
        <Menu
          display={false}
        >
          <StyledList
            active={false}
          >
            <StyledItem
              onClick={[Function]}
            >
              hello
            </StyledItem>
          </StyledList>
        </Menu>,
        <StyledButton
          onClick={[Function]}
          type="button"
        >
          Test
        </StyledButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "display": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <StyledList
            active={false}
          >
            <StyledItem
              onClick={[Function]}
            >
              hello
            </StyledItem>
          </StyledList>,
          "display": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": Array [
              <StyledItem
                onClick={[Function]}
              >
                hello
              </StyledItem>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": "hello",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "hello",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Test",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Test",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Background
            display={false}
            onClick={[Function]}
          />,
          <Menu
            display={false}
          >
            <StyledList
              active={false}
            >
              <StyledItem
                onClick={[Function]}
              >
                hello
              </StyledItem>
            </StyledList>
          </Menu>,
          <StyledButton
            onClick={[Function]}
            type="button"
          >
            Test
          </StyledButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "display": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <StyledList
              active={false}
            >
              <StyledItem
                onClick={[Function]}
              >
                hello
              </StyledItem>
            </StyledList>,
            "display": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": Array [
                <StyledItem
                  onClick={[Function]}
                >
                  hello
                </StyledItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "children": "hello",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "hello",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Test",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Test",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
